entry-asar/has-asar.js

```diff
@@ -1,7 +1,24 @@
 if (process.arch === 'arm64') {
-  process._archPath = require.resolve('../app-arm64.asar');
+  setPaths('arm64');
 } else {
-  process._archPath = require.resolve('../app-x64.asar');
+  setPaths('x64');
+}
+
+function setPaths(platform) {
+  // This should return the full path, ending in something like
+  // Notion.app/Contents/Resources/app.asar
+  const appPath = app.getAppPath()
+  const asarFile = `app-${platform}.asar`
+
+  // Maybe we'll handle this in Electron one day
+  if (appPath.includes('app.asar')) {
```

Comment:
I'm gonna sound paranoid, but can we do better here, some muppet somewhere is gonna make their username `app.asar` and complain that stuff broke.

Maybe instead of checking app path we check explicitly that it **ends** with `${path.sep}app.asar`.  Or just do `path.basename(thing) === 'app.asar'`

Then instead of string replacement to `path.join(path.dirname(thing), asarFile)`